APE_Black_Formatting.rst

```diff
@@ -0,0 +1,306 @@
+
+Migration effort
+~~~~~~~~~~~~~~~~
+Enthusiastic champions of the effort will largely do the work, but we recognize
+and appreciate that additional effort will be required from maintainers that are
+neutral or against the adoption of Black.
+
+Backports / release branches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This is addressed in the Implementation section.
+
+New contributors
+~~~~~~~~~~~~~~~~
+New contributors may be more discouraged by more steps. However, this *replaces*
+the existing PEP8 check, so that it is not "another step". To make it better,
+leave very clear explicit messages in the status checks on GitHub to reduce the
+barrier. Scikit-learn had a few issues with contributors but not at a level that
+caused regret about the decision to adopt black formatting. One developer
+described the change as a "net benefit".
+
+Removing the humanity
+~~~~~~~~~~~~~~~~~~~~~
+Some developers may feel that individual style and humanity gets lost in
+autoformatting. We hope to convey that humanity and individuality is expressed
+more fully in the elegance and clarity of the code logic and implementation. In
+most cases the placement of white space and line breaks can be viewed as a
+mechanistic process that humans need not control.
+
+Exception Policy
+^^^^^^^^^^^^^^^^
+As described above, there are cases Black formatting makes the code
+substantially more difficult to read or understand. These should be treated
+explicitly as exceptions, using the ``#fmt: off``/``#fmt: on`` escape hatch for
+multi-line exceptions, or ``#fmt: skip`` for single-line exceptions. **This is
+allowed if the code author and PR reviewer(s) agree that the code line or block
+should be excluded from Black formatting.**
+
+
+Branches and pull requests
+--------------------------
+
+- `Add configuration for black autoformatter <https://github.com/astropy/astropy/pull/13253>`_
+- `Apply black to modeling <https://github.com/astropy/astropy/pull/13254>`_
+
+Implementation
+--------------
+
+All Python code in ``astropy`` is formatted with Black_, using its default
+settings, that is, 88 characters per line and double quotes. Note that, by
+design, most of Black_ is inflexible, and there is not a way to pick-and-choose
+some elements and not others of Black's format.
+
+Implementing this change requires:
+
+1. Updating the `coding style`_:
+
+   * Adding documentation about Black, with a particular emphasis on the
+     rationale expressed here and the exception policy described above.
+   * Document the Black exception guidelines (as described in this APE)
+   * Removing other references to code formatting in the documentation.
+
+2. Updating flake8 configuration to be compatible with Black.
+
+   This is straightforward and well documented. It may even be removed if it
+   does not provide any added checks that ``black –check`` does not.
+
+3. Checking Black in CI.
+
+   * Add a CI check that runs ``black --check`` on the whole code base, failing
+     if it meets a violation
```

Comment:
This is potentially conflicting with the new use of pre-commit for these style checks as I believe it only checks the diffs. I personally don't really see a reason why we shouldn't continue to use pre-commit & pre-commit.ci so I propose the following:

```suggestion
   * Add a CI check that runs black to ensure consistency, failing if it meets a violation.
```

Being overly prescriptive here is not needed.

Reply:
That is what I wanted to ensure was possible with this wording. We will be running black as part of the CI (in pre-commit) so imo my suggestion means this APE allows us to do the easiest thing.